root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -76,16 +76,17 @@ def start_discovery_service
 end
 
 def configure_network static, mac, ip=nil, gw=nil, dns=nil, vlan=nil
-  command("systemctl stop foreman-proxy")
+  command("systemctl stop foreman-proxy", false)
   if static
     command("nm-configure primary-static '#{mac}' '#{ip}' '#{gw}' '#{dns}' '#{vlan}'")
   else
     command("nm-configure primary '#{mac}' '#{vlan}'")
   end
-  result = command("nmcli connection up primary", false)
```

Comment:
Note the "up" command is also not necessary at all, everything is automatic.

Unfortunately we are not reliably tell here if configuration was success or not, NM does the job and `nm-online` usually returns 0 very quickly (when NM daemon is up and at least one "must-not-fail" interface is starting = success).